test/test_assemblers.py

```diff
@@ -135,9 +136,17 @@ def nbd_connect(image):
         r = subprocess.run(["qemu-nbd", "--connect", device, "--read-only", image], check=False).returncode
         if r == 0:
             try:
+                # qemu-nbd doesn't wait for the device to be ready
+                while True:
```

Comment:
I feel like there should be a reasonable timeout so that we don't hang up the process in case something goes wrong on nbd side.

Reply:
For running them locally? (On CI we already have a global timeout.)

Reply:
The CI has timeout of 10 minutes, I think it's safe to assume the nbd is ready in 1 minute, so theoretically we can shorten the time to fail. However, I think it's such a rare case it doesn't make sense to make the code more complicated. Just merge it. :) I'm just a bit suspicious of loops that might be endless. :)

Reply:
You're right, endless loops are bad in itself. (I don't concede to your other points :stuck_out_tongue: )

Fixed. Thanks for the review!